dygraphs/dygraphs.d.ts

```diff
@@ -0,0 +1,1272 @@
+     */
+    drawHighlightPointCallback?: (
+      g: Dygraph,
+      seriesName: string,
+      canvasContext: CanvasRenderingContext2D,
+      cx: number,
+      cy: number,
+      color: string,
+      pointSize: number) => any;
+
+    /**
+     * Draw a custom item when drawPoints is enabled. Default is a small dot matching the series
+     * color. This method should constrain drawing to within pointSize pixels from (cx, cy).
+     * Also see <a href='#drawHighlightPointCallback'>drawHighlightPointCallback</a>
+     */
+    drawPointCallback?: (
+      g: Dygraph,
+      seriesName: string,
+      canvasContext: CanvasRenderingContext2D,
+      cx: number,
+      cy: number,
+      color: string,
+      pointSize: number) => any;
+
+    /**
+     * Does the data contain standard deviations? Setting this to true alters the input format.
+     */
+    errorBars?: boolean;
+
+    /**
+     * Sets the data being displayed in the chart. This can only be set when calling
+     * updateOptions; it cannot be set from the constructor. For a full description of valid data
+     * formats, see the <a href='http://dygraphs.com/data.html'>Data Formats</a> page.
+     */
+    file?: DataArray|GVizDataTable|string;
+
+    /**
+     * When set, attempt to parse each cell in the CSV file as "a/b", where a and b are integers.
+     * The ratio will be plotted. This allows computation of Wilson confidence intervals (see
+     * below).
+     */
+    fractions?: boolean;
+
+    /**
+     * Height, in pixels, of the chart. If the container div has been explicitly sized, this will
+     * be ignored.
+     */
+    height?: number;
+
+    /**
+     * Whether to hide the legend when the mouse leaves the chart area.
+     */
+    hideOverlayOnMouseOut?: boolean;
+
+    /**
+     * When set, this callback gets called every time a new point is highlighted.
+     */
+    highlightCallback?: (
+      event: MouseEvent,
+      xval: number,
+      points: Point[],
+      row: number,
+      seriesName: string) => any;
+
+    /**
+     * Fade the background while highlighting series. 1=fully visible background (disable
+     * fading), 0=hiddden background (show highlighted series only).
+     */
+    highlightSeriesBackgroundAlpha?: number;
+
+    /**
+     * When set, the options from this object are applied to the timeseries closest to the mouse
+     * pointer for interactive highlighting. See also 'highlightCallback'. Example:
+     * highlightSeriesOpts: { strokeWidth: 3 }.
+     */
+    highlightSeriesOpts?: PerSeriesOptions;
+
+    /**
+     * Usually, dygraphs will use the range of the data plus some padding to set the range of the
+     * y-axis. If this option is set, the y-axis will always include zero, typically as the
+     * lowest value. This can be used to avoid exaggerating the variance in the data
+     */
+    includeZero?: boolean;
+
+    /**
+     * TODO(konigsberg): document this
```

Comment:
lol, that's been a TODO for over five years :)
https://github.com/danvk/dygraphs/commit/74a5af31a87b245b89337037bc33a0e900153eae#diff-7d0b3e67115199dd9e5a0be36357dac9R180

I removed this comment since it's not useful in this context.